src/legacy/core_plugins/telemetry/server/collectors/application_usage/index.test.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UsageCollectionSetup } from '../../../../../../plugins/usage_collection/server';
+import { savedObjectsRepositoryMock } from '../../../../../../core/server/mocks';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { CollectorOptions } from '../../../../../../plugins/usage_collection/server/collector/collector';
+
+import { registerApplicationUsageCollector } from './';
+import {
+  ROLL_INDICES_INTERVAL,
+  SAVED_OBJECTS_TOTAL_TYPE,
+} from './telemetry_application_usage_collector';
+
+describe('telemetry_application_usage', () => {
+  jest.useFakeTimers();
+
+  let collector: CollectorOptions;
+
+  const usageCollectionMock: jest.Mocked<UsageCollectionSetup> = {
+    makeUsageCollector: jest.fn().mockImplementation(config => (collector = config)),
+    registerCollector: jest.fn(),
+  } as any;
+
+  const getUsageCollector = jest.fn();
+  const callCluster = jest.fn();
+
+  beforeAll(() => registerApplicationUsageCollector(usageCollectionMock, getUsageCollector));
+  afterAll(() => jest.clearAllTimers());
+
+  test('registered collector is set', () => {
+    expect(collector).not.toBeUndefined();
+  });
+
+  test('if no savedObjectClient initialised, return undefined', async () => {
+    expect(await collector.fetch(callCluster)).toBeUndefined();
+    jest.runTimersToTime(ROLL_INDICES_INTERVAL);
+  });
+
+  test('when savedObjectClient is initialised, return something', async () => {
```

Comment:
nit: We should also be testing internal methods used when we are returning the 'something'. i.e: `rollTotals`.